pkg/declarativeconfig/configuration.go

```diff
@@ -20,6 +20,8 @@ const (
 	RoleConfiguration          ConfigurationType = "role"
 )
 
+const supportedTypes = AuthProviderConfiguration + "," + AccessScopeConfiguration + "," + PermissionSetConfiguration + "," + RoleConfiguration
```

Comment:
What I thought about here is rather `const map[string]func() Configuration` and `supportedTypes` string is built on `init` out of map keys. Perhaps it's overkill, although thinking about the future to have one place in the file where you need to make changes to add a new type can be beneficial. 

Either way, I'll approve and leave the decision of whether to roll back or change to `const map[string]func() Configuration` to you.

Reply:
Hm, I'm not sure it makes a difference though? You'd still need to include a configuration type twice: first you'd have to define and export it, then you have to add it either to an init function or the helper function I had initially. I don't really see any need to introduce the init function here to be honest, but maybe I'm missing something 🤷